### DISCUSSION TITLE
Good practices for live demo'ing during development?

### DISCUSSION THREAD

**Discussion by collinskatie** [empiricaly/empirica#511] [Q&A] [answered]
Hi, I'm new to empirica, but it's awesome so far! I wanted to scope out what good practices are for demoing your experiment during development?

I'd like to be able to make a change and see how it impacts the interface/control flow. 

Right now, I've just kept empirica running in the command line, and then make a new batch, duplicating my earlier one. Before opening localhost:3000 in a separate tab for two players. However, this is leading to tons of batches being spawned! (see attached) It seems I can't keep a single batch open... is there a better way to just have a link that can be refreshed, without making a new batch each time? 

Thanks!

<img width="1302" alt="Screenshot 2024-02-22 at 20 45 33" src="https://github.com/empiricaly/empirica/assets/37914552/89bd2d07-2447-430d-8ff3-1404a56578e8">

**Comment by npaton** [comment, marked as answer]
Hey @collinskatie. If you are moving through the stages of your game while testing, unfortunately, there is no way at the moment to reverse that, and you'll need to create a new batch/game.

There are a few things you can do to help you with this:
- You can reset the players ui by clicking the "refresh" button (a circular arrow) on the bottom left of the player screen. This will save you from closing and reopening tabs. You will still require you to login with your player, with a new unique "id".
- Usually, I like to work on one stage at a time, at least when working on the UI, so during development, I tend to set the stage durations to a very large number (e.g. 1000000 seconds)
- If I'm testing logic between rounds/stages (callbacks), I like to also increase the number of rounds/stages, so I can modify the code between each stage submit
- Note that if the amount of Batches in the admin UI is becoming unwieldy, you can clear your data by stopping empirica, deleting `.empirica/local/tajriba.json` and restarting empirica. 

These crutches don't always work, depending on what you're trying to do. I would be interested in hearing everyone's ideas on how we might improve the development cycles. It is something I would love to improve.

Also, I'm now realizing, I hope I understood your question correctly, let me know.

**Comment by collinskatie** [comment]
Thank you @npaton !! We've been using the long time duration and refresh as you suggested. Working great! Thanks!
